sky/resources.py

```diff
@@ -853,6 +853,8 @@ def make_deploy_variables(
     def get_reservations_available_resources(
             self, specific_reservations: Set[str]) -> Dict[str, int]:
         """Returns the number of available reservation resources."""
+        if self.use_spot:
```

Comment:
Since we do it in this module, maybe add `# GCP's & AWS's reservations do not support spot instances. We assume other clouds behave the same.`

Reply:
Good point! Added the following comment.
```
# GCP's & AWS's reservations do not support spot instances. We
# assume other clouds behave the same. We can move this check down
# to each cloud if any cloud supports reservations for spot.
```